tests/test_writer.py

```diff
@@ -91,17 +87,57 @@ def test_writer_operations():
 
     writer.add_attachment("foobar.gif", b"foobarcontent")
 
-    # finally, write "output" to PyPDF2-output.pdf
-    tmp_path = "dont_commit_writer.pdf"
-    with open(tmp_path, "wb") as output_stream:
-        writer.write(output_stream)
-
     # Check that every key in _idnum_hash is correct
     objects_hash = [o.hash_value() for o in writer._objects]
     for k, v in writer._idnum_hash.items():
         assert v.pdf == writer
         assert k in objects_hash, "Missing %s" % v
 
+
+tmp_path = "dont_commit_writer.pdf"
+
+
+def test_writer_operations_by_totally_traditional_usage():
+    writer = PdfWriter()
+
+    writer_operate(writer)
+
+    # finally, write "output" to PyPDF2-output.pdf
+    with open(tmp_path, "wb") as output_stream:
+        writer.write(output_stream)
+
+    # cleanup
+    os.remove(tmp_path)
```

Comment:
as a point of interest, if something does error within this test, do we want this file to be left behind under some circumstance?

Reply:
For the writer, I don't think so. Is there a better way to do the cleanup?

For some other tests I sometimes look at the output (e.g. for the annotations). I want to see that it looks right; so during development I comment the cleanup out.

Reply:
I suggest using a [`tempfile.TemporaryDirectory()`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) class. There's also `TemporaryFile` if you want to act directly on a file-like object, but it has caveats around opening it multiple times, so may not be suitable, vs `TemporaryDirectory` which just gives us a new empty directory we can create whatever files into.

Something like:
```python
import tempfile
with tempfile.TemporaryDirectory() as tmp:
    tmp_path = os.path.join(tmp, 'file.pdf')
    writer = PdfWriter()

    writer_operate(writer)

    # finally, write "output" to PyPDF2-output.pdf
    with open(tmp_path, "wb") as output_stream:
        writer.write(output_stream)
```

where then on leaving the context manager, `TemporaryDirectory()` will automatically remove the directory.

Reply:
Thank you for the suggestion :+1: 

It feels to me that this makes the tests more noisy, but [the `tmp_path` fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture) seems to follow the same idea :-)

Reply:
I've added the fixture - thank you :hugs: